source/core/rheological_model.cc

```diff
@@ -8,7 +8,11 @@ RheologicalModel<dim>::model_cast(
   if (!physical_properties.non_newtonian_flow)
     return std::make_shared<Newtonian<dim>>(
       physical_properties.fluids[0].viscosity);
-  else //  (physical_properties.non_newtonian_parameters.model ==
+  else if (physical_properties.non_newtonian_parameters.model ==
+           Parameters::NonNewtonian::Model::powerlaw)
+    return std::make_shared<PowerLaw<dim>>(
+      physical_properties.non_newtonian_parameters);
+  else // if (physical_properties.non_newtonian_parameters.model ==
        // Parameters::NonNewtonian::Model::carreau)
     return std::make_shared<Carreau<dim>>(
       physical_properties.non_newtonian_parameters);
```

Comment:
It would make more sense to me if the Newtonian alternative was the default one, but if you prefer to have it first instead that also makes sense

Reply:
That's a good point, I'll change it :)